content/post/2019-01-20-what-is-production-grade.markdown

```diff
@@ -0,0 +1,270 @@
+---
+title: "What Is Production Grade?"
+date: 2019-01-20T15:35:03-08:00
+---
+
+I recently decided to try the Caddy webserver project's implementation
+of a webserver written in Golang.  It had many interesting things
+going for it, including automagic certificate renewal via ACME and
+Let's Encrypt, "plugins" written in a sane language, dramatically
+simplified configuration, and perhaps in one of the most boastful
+statements I've seen of any software recently, it claims to be
+"Production-Ready" (features page, right hand side).  Given that it
+took me the better part of a day to get a functional Caddy role for
+Ansible, I really question this claim.
+
+So that's what this article is about.  I'm putting down here what I
+think production grade software should be, and some of the problems
+with what software is out there.
+
+## What is Production?
+
+To answer the original question, we have to first look into what is
+production.  In the general sense, production is agreed to be the
+environment of a software system that is either directly or
+transitively involved in serving end-users.  These end-users can be
+internal or external to an organization, but the defining
+characteristic that makes someone an end user is that they can be
+unhappy about the status of a system and their unhappiness directly
+translates into a need to act for some team responsible for the
+system.  This is in contrast to dev or QA environments where
+unhappiness about the system state does not demand action.
```

Comment:
I'd add immediate before action. A dev environment broken for months isn't tenable.